common/dbus/platform/asio_libdbus/dbus_client.cpp

```diff
@@ -437,6 +493,13 @@ DBusHandlerResult MsgFilter(DBusConnection *connection, DBusMessage *message, vo
 	const string signature {dbus_message_get_signature(message)};
 
 	auto opt_string_handler = client->GetSignalHandler<expected::ExpectedString>(spec);
+	auto opt_string_pair_handler = client->GetSignalHandler<ExpectedStringPair>(spec);
+
+	// either no match or only one match
+	assert(
+		!(static_cast<bool>(opt_string_handler) || static_cast<bool>(opt_string_pair_handler))
```

Comment:
Q: Why cast to `bool`, when it's an `optional` already?

Reply:
So, multiple matches is a programming error?

Reply:
Because otherwise it fails to compile with `optional` not having `operator||` and `operator^`.

Reply:
`(opt_string_handler || opt_string_pair_handler)` compiles just fine

Reply:
For `^` I was getting a compilation error.